mq/plugins/remove_dots.py

```diff
@@ -0,0 +1,35 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+import json
+
+
+class message(object):
+
+    def __init__(self):
+        '''
+        takes an incoming message
+        and checks for dots at the
+        start or end of the key and
+        removes them
+        '''
+
+        self.registration = ['cloudtrail']
+        self.priority = 5
+
+    def onMessage(self, message, metadata):
+        def remove_dots(obj):
+            for key in obj.keys():
+                if key[0] == '.' or key[-1] == '.':
+                    new_key = key.replace(".","")
+                    if new_key != key:
+                        obj[new_key] = obj[key]
+                        del obj[key]
+            return obj
+
+        new_json = json.loads(json.dumps(message), object_hook=remove_dots)
```

Comment:
I'm a bit worried that doing both a dumps and a loads operation is a bit unnecessary and might cause performance problems. Instead of using json, would it be better to recursively loop through all the keys, similar to https://github.com/mozilla/MozDef/blob/master/mq/plugins/lower_keys.py . Thoughts?

Reply:
acknowledged, pinging you on slack

Reply:
I've  rewritten this code, I think I overcomplicated it when I originally attempted to make it similar to lower_keys, which is why I ended up with the dumps and loads.
This wasn't nearly as complicated this time around. Thank you!